### DISCUSSION TITLE
how to debug CodegenC.tpl or CodegenC.mo file step by step?

### DISCUSSION THREAD

**Discussion by pily1** [OpenModelica/OpenModelica#12641] [Q&A] [answered]
Hi, I want to debug step by step to watch the input value and output value of a function in the 'CodegenC.mo' file. How can I do it?

**Comment by AnHeuermann** [comment]
Oh that's more or less impossible. You can use gdb to step through the generated C sources but that's about it.

**Reply by pily1** [reply]
When I use gdb to step through the generated C source code, I find that all parameters are of type void*, and I can't get the true value to understand them。

**Reply by AnHeuermann** [reply, marked as answer]
Yes, it's basically pointless without casting to the boxed memory types. You could try the MDT debugger with Exclipse: https://github.com/OpenModelica/OpenModelica/wiki/Setting-up-MDT-debugger-in-Eclipse. It can do this (sometimes, mostly it will crash...)

**Reply by pily1** [reply]
think you very much,I have installed MDT failed,I will try print everything solution
